### PR TITLE
Hpo

### PR DESCRIPTION
**Pull request by simonschuang**

### CODE REVIEW

**Comment by simonschuang** [comment]
The original validation output is
```
Validation (epoch 1): accuracy = 0.14, top5_accuracy = 0.21
Training (epoch 1.1): loss = 2.3, accuracy = 0.21, lr = 0.1
```
If HPO is enabled, to display multiple lines in same graph,
the training and validation outputs should have prefix `runX_`
```
Validation (epoch 1): run1_accuracy = 0.14, run1_top5_accuracy = 0.21
Training (epoch 1.1): run1_loss = 2.3, run1_accuracy = 0.21, run1_lr = 0.1
```

**Comment by simonschuang** [comment]
During HPO training, each run results should be stored on different subdirectory of training job dir.
ex. 
 /data/20180723-000001/run1/checkpoint-xxx
 /data/20180723-000001/run2/checkpoint-xxx
 /data/20180723-000001/run3/checkpoint-xxx
And, after HPO runs. Copy checkpoint files of the best accuracy run to the job directory
/data/20180723-000001/checkpoint-xxx

**Comment by simonschuang** [comment]
For consistency, please refactor code to move initialize variables code under Benchmark() __init__()  here https://github.com/myelintek/web/blob/hpo/digits/tools/tf/trainer.py#L355
initialize variables code including 
```
FLAGS.piecewise_learning_rate_schedule
FLAGS.hpo
FLAGS.num_bayesian_runs
```

Also, can we do the bayesian_runs loop here? 
https://github.com/myelintek/web/blob/hpo/digits/tools/tf/trainer.py#L890-L891
